[docs] Improve and update Fonts guide

amandeepmittal opened the pull request:
# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.
-->

This PR improves the Fonts guide under the Develop section. The changes follow-up #28804 and is in continuation to improve the guides under Develop section.

Closes ENG-12276

# How

<!--
How did you build this feature or fix this bug and why?
-->

- Restructure the guide to use a more "how-to" guide approach for using custom and Google Fonts
- Put all the additional/good-to-know information under Additional information section
- Provide examples at each step and update the existing ones accordingly
- Remove the outdated React Class component section and example for `Font.loadAsync`
- Define what's a PostScript name is and how to reference a font name in the project when using a local file or Google Fonts
- Add benefits and limitations to each approach (Embedding Local file with config plugin and using `useFonts` hook)
- Update internal links in Icons guide
- Update Redirects after changing the section name in Fonts guide
- Improve and update the example under the Usage section in Font API reference #29905. Changes applied to unversioned and SDK 51.

 

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

Run docs locally and visit: http://localhost:3002/develop/user-interface/fonts or See Preview: /develop/user-interface/fonts/.

# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

amandeepmittal committed:
[docs] Update Fonts guide in Develop section

amandeepmittal committed:
First draft done

amandeepmittal committed:
Additional information section Part 1

amandeepmittal committed:
Draft complete

amandeepmittal committed:
Update internal links, fix typos, add box links

amandeepmittal committed:
Remove unused image file

amandeepmittal committed:
Fix Icons guide formatting

amandeepmittal committed:
Update image path

amandeepmittal committed:
Merge branch 'main' into aman/update-fonts-guide

Simek commented on docs/pages/develop/user-interface/fonts.mdx:
```diff
@@ -59,323 +77,349 @@ Install [`expo-font`](/versions/latest/sdk/font/#installation) library and add t
+<Step label="1">
 
-For example, to use Inter font you can install the [`@expo-google-fonts/inter`](https://www.npmjs.com/package/@expo-google-fonts/inter) package with the command below.
+Install the font package. For example, to use Inter-Black font, install the [`@expo-google-fonts/inter`](https://www.npmjs.com/package/@expo-google-fonts/inter) package with the command below.
```
```suggestion
Install the font package. For example, to use Inter Black font, install the [`@expo-google-fonts/inter`](https://www.npmjs.com/package/@expo-google-fonts/inter) package with the command below.
```

Naming nit.

Simek commented on docs/pages/develop/user-interface/fonts.mdx:
```diff
@@ -59,323 +77,349 @@ Install [`expo-font`](/versions/latest/sdk/font/#installation) library and add t
-  let [fontsLoaded, fontError] = useFonts({
+<Step label="1">
+
+Instal the Google Fonts package, `expo-font` and `expo-splash-screen` libraries.
```
```suggestion
Install the Google Fonts package, `expo-font` and `expo-splash-screen` libraries.
```

Typo fix.

Simek commented on docs/pages/develop/user-interface/fonts.mdx:
````diff
@@ -59,323 +77,349 @@ Install [`expo-font`](/versions/latest/sdk/font/#installation) library and add t
+
+```tsx app/_layout.tsx
+// Rest of the import statements
+/* @info Import <CODE>useFonts</CODE> hook from <CODEexpo-font</CODE>. */ import { useFonts } from 'expo-font'; /* @end */
````
```suggestion
/* @info Import <CODE>useFonts</CODE> hook from <CODE>expo-font</CODE>. */ import { useFonts } from 'expo-font'; /* @end */
```

Typo fix.

Simek reviewed (approved):
Looks good! 👍 Left just few nits here and there.

amandeepmittal committed:
[docs] Update example snippet in Usage section in Font API reference (#29905)

# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub
issues, forums posts, or feature requests.
-->

There's a difference between how [Fonts guide shows the
example](https://docs.expo.dev/develop/user-interface/fonts/#minimal-example)
using a Custom Font vs how Font API reference does in Usage section.

We should keep the same pattern of using examples in both places so that
a developer can easily use it in their app, especially with the new
default template which uses `useEffect` instead of `useCallback`.

Follow-up #29903

# How

<!--
How did you build this feature or fix this bug and why?
-->

Update the example in Usage section of Font API reference.

Changes applied to `unversioned` and SDK 51 references.

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could
reproduce your test, especially if this PR does not include automated
tests! If possible, please also provide terminal output and/or
screenshots demonstrating your test/reproduction.
-->

By updating example snippets and then running Snack to test it.

![CleanShot 2024-06-20 at 20 28
29@2x](https://github.com/expo/expo/assets/10234615/66928778-d517-4338-a255-273ccd53c131)


# Checklist

<!--
Please check the appropriate items below if they apply to your diff.
This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg:
https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style
Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build
(eg: updated a module plugin).

amandeepmittal commented on docs/pages/guides/icons.mdx:
`````diff
@@ -202,13 +205,15 @@ export default function App() {
-    alignItems: 'center',
-  },
+container: {
+flex: 1,
+justifyContent: 'center',
+alignItems: 'center',
+},
 });
-```
+
+````
`````
Yep, fixing it.

amandeepmittal committed:
Apply feedback from Simek
